src/stan/mcmc/chains.hpp

```diff
@@ -254,22 +256,39 @@ namespace stan {
           var_plus += variance(chain_mean);
         Eigen::VectorXd rho_hat_t(n_samples);
         rho_hat_t.setZero();
-        double rho_hat = 0;
-        int max_t = 0;
-        for (int t = 1; (t < n_samples && rho_hat >= 0); t++) {
-          Eigen::VectorXd acov_t(chains);
-          for (int chain = 0; chain < chains; chain++) {
-            acov_t(chain) = acov(chain)(t);
+        double rho_hat_even = 1;
+        double rho_hat_odd = 0;
+        Eigen::VectorXd acov_t(chains);
+        for (int chain = 0; chain < chains; chain++)
+          acov_t(chain) = acov(chain)(1);
+        rho_hat_odd = 1 - (mean_var - mean(acov_t)) / var_plus;
+        rho_hat_t(1) = rho_hat_odd;
+        // Geyer's initial positive sequence
+        int max_t = 1;
+        for (int t = 1;
```

Comment:
Please switch the indexes to start at `t = 2`. If you look in the loop, there's no direct use of `t`, it's always `t + 1` or `t + 2`. (The only place it's used as `t` is within the condition in the for loop, which can be changed easily)

Reply:
Indexing was to follow BDA3 to make it easier to compare code and explanation. Due to adding monotone sequence, the indexing in manual is now different from BDA3. Should I follow your recommendation or try to make code readable compared to equations in documentation?

Reply:
I see. Let me look at BDA3 and I’ll get back to you.

Reply:
I think it's better to look at the updated stan/src/docs/stan-reference/inference.tex as it has now different indexing than BDA3. The initial positive sequence was easy enough to describe without having explicit symbol for the paired sums and Andrew wanted then leave that out from BDA3. Monotone sequence is easier to describe with explicit symbol for paired sums, and the updated inferences.tex uses that. The code is closer to description in inferences.tex but doesn't have the explicit variable for the paired sum, so the indexing is still different.